src/Storages/MergeTree/TemporaryParts.cpp

```diff
@@ -0,0 +1,24 @@
+#include <Storages/MergeTree/TemporaryParts.h>
+
+namespace DB
+{
+
+bool TemporaryParts::contains(const std::string & basename)
+{
+    std::lock_guard lock(mutex);
+    return parts.contains(basename);
+}
+
+void TemporaryParts::add(const std::string & basename)
+{
+    std::lock_guard lock(mutex);
+    parts.emplace(basename);
+}
```

Comment:
```suggestion
void TemporaryParts::add(std::string basename)
{
    std::lock_guard lock(mutex);
    parts.emplace(std::move(basename));
}
```
To make it more flexible if we decide to send an rvalue to it.

Reply:
Done.